### ISSUE TITLE
Unsupported code error

### ISSUE THREAD

**Issue by Hansrider** [shogo4405/HaishinKit.dart#1]
<img width="536" alt="image" src="https://user-images.githubusercontent.com/19708989/179546409-56893a9b-2b35-4624-9807-b5b14f14e10d.png">

```
Terminating app due to uncaught exception 'NSInternalInconsistencyException', reason: 'Unsupported value for standard codec'

Unsupported value: [] of type __SwiftValue
*** Assertion failure in -[FlutterStandardWriter writeValue:], FlutterStandardCodec.mm:338
*** Terminating app due to uncaught exception 'NSInternalInconsistencyException', reason: 'Unsupported value for standard codec'
*** First throw call stack:
(0x1b13c1288 0x1ca0bb744 0x1b2c4e360 0x10a673478 0x10a6735ac 0x10a6735ac 0x10a673b00 0x10a6712fc 0x105bc5c50 0x105bc5388 0x105bc5454 0x1b1352834 0x1b13eefd4 0x1b13c21d0 0x1b13688ac 0x1b2b37754 0x1065913bc 0x1065914c8 0x10665ccf0 0x106652038 0x106652644 0x106652644 0x106652788 0x10667efe0 0x10661afd0 0x10661bcac 0x10661c2f8 0x1b142c120 0x1b144917c 0x1b1026e6c 0x1b1028a30 0x1b1030124 0x1b1030c80 0x1b103b500 0x222c800bc 0x222c7fe5c)
libc++abi: terminating with uncaught exception of type NSException
* thread #34, queue = 'com.haishinkit.HaishinKit.NetSocket.input', stop reason = signal SIGABRT
    frame #0: 0x00000001e8f56b38 libsystem_kernel.dylib`__pthread_kill + 8
libsystem_kernel.dylib`__pthread_kill:
->  0x1e8f56b38 <+8>:  b.lo   0x1e8f56b58               ; <+40>
    0x1e8f56b3c <+12>: pacibsp 
    0x1e8f56b40 <+16>: stp    x29, x30, [sp, #-0x10]!
    0x1e8f56b44 <+20>: mov    x29, sp
Target 0: (Runner) stopped.
```

**Comment by shogo4405** [comment]
Can I see your dart code?

**Comment by Hansrider** [comment]
i just copy paste this your example and testing on iphone 12 PRO , os version : 15.5
//import 'dart:async';

import 'package:audio_session/audio_session.dart';
import 'package:flutter/material.dart';
import 'package:haishin_kit/audio_settings.dart';
import 'package:haishin_kit/audio_source.dart';
import 'package:haishin_kit/net_stream_drawable_texture.dart';
import 'package:haishin_kit/rtmp_connection.dart';
import 'package:haishin_kit/rtmp_stream.dart';
import 'package:haishin_kit/video_settings.dart';
import 'package:haishin_kit/video_source.dart';
import 'package:permission_handler/permission_handler.dart';

class Streaming extends StatefulWidget {
  const Streaming({Key? key}) : super(key: key);

  @override
  State<Streaming> createState() => _StreamingState();
}

class _StreamingState extends State<Streaming> {
  RtmpConnection? _connection;
  RtmpStream? _stream;
  bool _recording = false;
  String _mode = "publish";
  CameraPosition currentPosition = CameraPosition.back;

  @override
  void initState() {
    super.initState();
    initPlatformState();
  }

  @override
  void dispose() {
    _stream?.dispose();
    _connection?.dispose();
    super.dispose();
  }

  Future<void> initPlatformState() async {
    await Permission.camera.request();
    await Permission.microphone.request();

    // Set up AVAudioSession for iOS.
    final session = await AudioSession.instance;
    await session.configure(const AudioSessionConfiguration(
      avAudioSessionCategory: AVAudioSessionCategory.playAndRecord,
      avAudioSessionCategoryOptions:
          AVAudioSessionCategoryOptions.allowBluetooth,
    ));

    RtmpConnection connection = await RtmpConnection.create();
    connection.eventChannel.receiveBroadcastStream().listen((event) {
      switch (event["data"]["code"]) {
        case 'NetConnection.Connect.Success':
          if (_mode == "publish") {
            _stream?.publish("live");
          } else {
            _stream?.play("live");
          }
          setState(() {
            _recording = true;
          });
          break;
      }
    });

    RtmpStream stream = await RtmpStream.create(connection);
    stream.audioSettings = AudioSettings(muted: true, bitrate: 64 * 1000);
    stream.videoSettings = VideoSettings(
      width: 640,
      height: 360,
      bitrate: 160 * 1000,
      muted: true,
    );
    stream.attachAudio(AudioSource());
    stream.attachVideo(VideoSource(position: currentPosition));

    if (!mounted) return;

    setState(() {
      _connection = connection;
      _stream = stream;
    });
  }

  @override
  Widget build(BuildContext context) {
    return MaterialApp(
      home: Scaffold(
        appBar: AppBar(title: const Text('HaishinKit'), actions: [
          IconButton(
            icon: const Icon(Icons.play_arrow),
            onPressed: () {
              if (_mode == "publish") {
                _mode = "playback";
                _stream?.attachVideo(null);
                _stream?.attachAudio(null);
              } else {
                _mode = "publish";
                _stream?.attachAudio(AudioSource());
                _stream?.attachVideo(VideoSource(position: currentPosition));
              }
            },
          ),
          IconButton(
            icon: const Icon(Icons.flip_camera_android),
            onPressed: () {
              if (currentPosition == CameraPosition.front) {
                currentPosition = CameraPosition.back;
              } else {
                currentPosition = CameraPosition.front;
              }
              _stream?.attachVideo(VideoSource(position: currentPosition));
            },
          )
        ]),
        body: Center(
          child: _stream == null
              ? const Text("")
              : NetStreamDrawableTexture(_stream),
        ),
        floatingActionButton: FloatingActionButton(
          child: _recording
              ? const Icon(Icons.fiber_smart_record)
              : const Icon(Icons.not_started),
          onPressed: () {
            if (_recording) {
              _connection?.close();
              setState(() {
                _recording = false;
              });
            } else {
              _connection?.connect(
                  "rtmp://ex-hpr-tokam-ams-eaa301568085045e.elb.eu-west-1.amazonaws.com/LiveApp/arnaud");
            }
          },
        ),
      ),
    );
  }
}

**Comment by shogo4405** [comment]
Oh, thank you. AWS response incompatible FlutterStandardCodec.mm. 
Think for a moment. Thank you.

> Optional(Optional(["capabilities": Optional(33.0), "mode": Optional(1.0), "level": Optional("status"), "code": Optional("NetConnection.Connect.Success"), "fmsVer": Optional("RED5/1,0,9,0"), "description": Optional("Connection succeeded."), "data": Optional([])]))
Unsupported value: [] of type __SwiftValue

**Comment by Hansrider** [comment]
ok,  what can i do to fix it ? or can you do something ?

**Comment by shogo4405** [comment]
No. Please wait. I have to fix this issue.

**Comment by Hansrider** [comment]
OK thank you for your reply,  I have another question , is it possible to specify the stream id and the stream name ? 

i have tested your plugin in Android OS and it is working but my rtmp are stopped every 41 secondes with this error 

<img width="920" alt="image" src="https://user-images.githubusercontent.com/19708989/179922296-3206e28d-33e5-4516-9719-d441e24e7321.png">

**Comment by alexeyslesarenko** [comment]
The problem is still reproducable in 0.9.2 version.

`2022-11-25 15:58:32.967643+0300 Runner[46477:3428680] *** Terminating app due to uncaught exception 'NSInternalInconsistencyException', reason: 'Unsupported value for standard codec'
*** First throw call stack:
(
	0   CoreFoundation                      0x00000001121d2378 __exceptionPreprocess + 242
	1   libobjc.A.dylib                     0x00000001112c2baf objc_exception_throw + 48
	2   Foundation                          0x0000000112c4d6ac _userInfoForFileAndLine + 0
	3   Flutter                             0x000000011bb95c85 -[FlutterStandardWriter writeValue:] + 1892
	4   Flutter                             0x000000011bb95bd9 -[FlutterStandardWriter writeValue:] + 1720
	5   Flutter                             0x000000011bb95bd9 -[FlutterStandardWriter writeValue:] + 1720
	6   Flutter                             0x000000011bb949bb -[FlutterStandardMethodCodec encodeSuccessEnvelope:] + 103
	7   Flutter                             0x000000011bb936eb ___ZL39SetStreamHandlerMessageHandlerOnChannelPU31objcproto20FlutterStreamHandler8NSObjectP8NSStringPU33objcproto22FlutterBinaryMessengerS_PU29objcproto18FlutterMethodCodecS_PU27objcproto16FlutterTaskQueueS__block_invoke_2 + 147
	8   haishin_kit                         0x000000010ffe2bae $syXlSgIeyBy_ypSgIegn_TR + 190
	9   haishin_kit                         0x000000010ffe239c $s11haishin_kit21RTMPConnectionHandlerC7handler33_78F86EF804834F9F2C2460647B3E6A45LLyy10Foundation12NotificationVF + 684
	10  haishin_kit                         0x000000010ffe245f $s11haishin_kit21RTMPConnectionHandlerC7handler33_78F86EF804834F9F2C2460647B3E6A45LLyy10Foundation12NotificationVFTo + 111
	11  CoreFoundation                      0x00000001121035db __CFNOTIFICATIONCENTER_IS_CALLING_OUT_TO_AN_OBSERVER__ + 137
	12  CoreFoundation                      0x0000000112103511 ___CFXRegistrationPost_block_invoke + 86
	13  CoreFoundation                      0x0000000112102af0 _CFXRegistrationPost + 541
	14  CoreFoundation                      0x00000001121023f9 _CFXNotificationPost + 822
	15  Foundation                          0x0000000112ccf604 -[NSNotificationCenter postNotificationName:object:userInfo:] + 82
	16  HaishinKit                          0x000000010e541388 $s10HaishinKit15EventDispatcherC8dispatch5eventyAA0C0C_tF + 1368
	17  HaishinKit                          0x000000010e5414e3 $s10HaishinKit15EventDispatcherC8dispatch_7bubbles4datayAA0C0C4NameV_SbypSgtF + 179
	18  HaishinKit                          0x000000010e617485 $s10HaishinKit18RTMPCommandMessageC7execute_4typeyAA14RTMPConnectionC_AA13RTMPChunkTypeOtF + 1733
	19  HaishinKit                          0x000000010e60bf61 $s10HaishinKit14RTMPConnectionC6listenyy10Foundation4DataVF + 4305
	20  HaishinKit                          0x000000010e60c157 $s10HaishinKit14RTMPConnectionC6listenyy10Foundation4DataVF + 4807
	21  HaishinKit                          0x000000010e60c157 $s10HaishinKit14RTMPConnectionC6listenyy10Foundation4DataVF + 4807
	22  HaishinKit                          0x000000010e60c157 $s10HaishinKit14RTMPConnectionC6listenyy10Foundation4DataVF + 4807
	23  HaishinKit                          0x000000010e60c157 $s10HaishinKit14RTMPConnectionC6listenyy10Foundation4DataVF + 4807
	24  HaishinKit                          0x000000010e60c849 $s10HaishinKit14RTMPConnectionCAA18RTMPSocketDelegateA2aDP6listenyy10Foundation4DataVFTW + 9
	25  HaishinKit                          0x000000010e63b845 $s10HaishinKit10RTMPSocketC6listenyyF + 2005
	26  HaishinKit                          0x000000010e5d27ee $s10HaishinKit9NetSocketC7doInput33_2F19F15DB5A1ACFB712D8A74CB6C271FLLyySo13NSInputStreamCF + 2110
	27  HaishinKit                          0x000000010e5d372b $s10HaishinKit9NetSocketC6stream_6handleySo8NSStreamC_So0G5EventVtF + 1339
	28  HaishinKit                          0x000000010e5d3fbd $s10HaishinKit9NetSocketC6stream_6handleySo8NSStreamC_So0G5EventVtFTo + 61
	29  CoreFoundation                      0x0000000112155522 _signalEventSync + 219
	30  CoreFoundation                      0x0000000112155569 ___signalEventQueue_block_invoke + 18
	31  libdispatch.dylib                   0x0000000116154d18 _dispatch_call_block_and_release + 12
	32  libdispatch.dylib                   0x0000000116155f5b _dispatch_client_callout + 8
	33  libdispatch.dylib                   0x000000011615dd41 _dispatch_lane_serial_drain + 1064
	34  libdispatch.dylib                   0x000000011615eb1b _dispatch_lane_invoke + 441
	35  libdispatch.dylib                   0x000000011616cf13 _dispatch_workloop_worker_thread + 959
	36  libsystem_pthread.dylib             0x00000001163b6ce3 _pthread_wqthread + 326
	37  libsystem_pthread.dylib             0x00000001163b5c67 start_wqthread + 15
)
libc++abi: terminating with uncaught exception of type NSException
Message from debugger: Terminated due to signal 6`

**Comment by shogo4405** [comment]
fixed dc1a2ae4f233d52881d5cb481909aec4c6b2fc75
> The problem is still reproducable in 0.9.2 version.
> 
> ```
> 2022-11-25 15:58:32.967643+0300 Runner[46477:3428680] *** Terminating app due to uncaught exception 'NSInternalInconsistencyException', reason: 'Unsupported value for standard codec'
> ```

**Comment by shogo4405** [comment]
fixed https://github.com/shogo4405/HaishinKit.dart/issues/1#issuecomment-1189924853 https://github.com/shogo4405/HaishinKit.kt/commit/646b9730fccb86157672d6591efdc9f6fc741ec4
